plugins/bulk-import-backend/src/service/githubApiService.ts

```diff
@@ -44,18 +45,31 @@ import { DefaultPageNumber, DefaultPageSize } from './handlers/handlers';
 
 const GITHUB_DEFAULT_API_ENDPOINT = 'https://api.github.com';
 
+// Cache size and TTL, based on the lower values of rate limits imposed by GH,
+// i.e., 5K requests per hour for requests using a personal token.
+// GitHub Apps owned by enterprises have a higher limit of 15K per hour.
+// See https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28
+const RESPONSE_CACHE_MAX_ITEMS = 5000;
+const RESPONSE_CACHE_TTL_MINUTES = 60;
+
 export class GithubApiService {
   private readonly logger: Logger;
   private readonly integrations: ScmIntegrations;
   private readonly githubCredentialsProvider: CustomGithubCredentialsProvider;
   private readonly config: Config;
+  // In-memory cache for storing ETags (used for efficient caching of unchanged data returned by GitHub)
+  private readonly responseCache: LRUCache<string, any>;
```

Comment:
I don't know if it would be better or not but have you considered using the built in [CacheService](https://backstage.io/docs/backend-system/core-services/cache/) of Backstage? I know for the RBAC plugin, we ran into an issue with scaling whenever we used in memory cache.

Reply:
Oh TIL there was a built-in cache service (which also allows to set a TTL, which is what I was mostly looking for). I'll look into using that, and remove the dependency to `lru-cache` if it works. Thanks @PatAKnight!

Reply:
PTAL at [`7a31008` (#2137)](https://github.com/janus-idp/backstage-plugins/pull/2137/commits/7a310088f7df62e6bbb3fdb4b9a75cc0578e68b0).
The built-in CacheService seems to work fine. I noticed that it can be backed either by an in-memory Map or even a Redis store, which is great. However, with the default in-memory store, I am not sure how it would scale since there is no max size (in case unexpired elements keep being added to the cache). Anyway, I think it should be good for now. I guess that would be something to keep an eye out on.